src/operator/tensor/elemwise_unary_op_basic.cc

```diff
@@ -1117,15 +1117,15 @@ MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU_DR(_backward_log10,
                                                   unary_bwd<mshadow_op::log10_grad>)
 .set_attr<nnvm::FGradient>("FGradient",
   [](const nnvm::NodePtr& n, const std::vector<nnvm::NodeEntry>& ograds) {
-    // ograds[0]: dL/dxgrad
+    // ograds[0]: dL/dygrad
```

Comment:
I think this is dL/dx_grad.  The head gradient is the gradient with respect to the previous output right? the previous output is x_grad  or dL/dx     so this thing is dL/(dL/dx)  or dL/dx_grad  in lack of a better notation.

Reply:
I guess it should be, dL/dy_grad as we are computing/returning dL/dx_grad,
Eg.
```
y = f(dx_grad)
L = g(y) # dx_grad formed part of the network and affected loss

During backprop by chain rule,
dL/dx_grad = dL/dy * dy/dx_grad

In comments, we have called dL/dy (mentioned in the above example) as dL/dy_grad
```

That is why we have,
https://github.com/apache/incubator-mxnet/blob/5b95fb3ee3581ba20fe1def336621d68a811e17f/src/operator/tensor/elemwise_unary_op_basic.cc#L1111-L1112

These multiplications performing, 
```
dL/dx_grad = dL/dy * dy/dx_grad
```

Reply:
I think the notation is complicating us in excess as it gets pretty hairy.  It's the head gradient of the previous (output) node, which has shape of x, and x_grad. So it has to be related to x, not y.

Reply:
Oh. I get it now. If I understand it correctly then, crudely `ograds[0]` is how much does the `x_grad` affect the `L` and then we compute how does `x_grad` change with `x`. Makes sense now. 

Thank you very much. Will reflect it in this and other PRs.

Reply:
@kshitij12345 I think what you write makes sense. I'm also unsure about notations, maybe you can come with a better one. If not maybe we leave the comment out, so we can merge the PR, as the code seems to do what's needed.

Reply:
Sure. Thanks Again.